front/components/vaults/VaultSideBarMenu.tsx

```diff
@@ -431,33 +433,87 @@ const VaultDataSourceViewItem = ({
   item,
   owner,
   vault,
+  node,
 }: {
   item: DataSourceViewType;
   owner: LightWorkspaceType;
   vault: VaultType;
+  node?: DataSourceViewContentNode;
 }): ReactElement => {
   const router = useRouter();
+  const [isExpanded, setIsExpanded] = useState(false);
 
-  const LogoComponent = getConnectorProviderLogoWithFallback(
-    item.dataSource.connectorProvider,
-    FolderIcon
-  );
-  const dataSourceViewPath = `/w/${owner.sId}/vaults/${vault.sId}/categories/${item.category}/data_source_views/${item.sId}`;
+  const { isNodesLoading, nodes } = useDataSourceViewContentNodes({
+    dataSourceView: item,
+    owner,
+    parentId: node?.internalId,
+    viewType: "documents",
+  });
+
+  // Load the currently selected node from router.query.parentId
+  const {
+    nodes: [selected],
+  } = useDataSourceViewContentNodes({
```

Comment:
Is this call cached wrt to the main view?

It's an expensive one? What's the sorry for pagination here?

Reply:
It's cached by swr, so done only once per page (or when revalidating). It's just a single content-node fetch (no children), so quite fast

Reply:
To be clear - beside this query, we have a lot of content-node queries which are done , one for each item in the tree, so it clearly increases the load. But if we manage to leverage swr caching correctly this will also fix  #1339 .